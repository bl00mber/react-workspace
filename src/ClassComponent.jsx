class ClassComponent extends React.Component {
  render() {
    return <h3>{this.props.name}</h3>;
  }
}
