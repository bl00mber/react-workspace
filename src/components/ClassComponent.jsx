import React from 'react';
import PropTypes from 'prop-types';

class ClassComponent extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      test: true,
    };
  }

  render() {
    return this.state.test && <h3>{this.props.name}</h3>;
  }
}

ClassComponent.propTypes = {
  name: PropTypes.string.isRequired,
};

export default ClassComponent;
