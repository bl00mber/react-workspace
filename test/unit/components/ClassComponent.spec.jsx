import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ClassComponent from '@/components/ClassComponent';

describe('ClassComponent', () => {
  test('should match enzyme shallow snapshot', () => {
    const wrapper = shallow(<ClassComponent name="Class Component" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
