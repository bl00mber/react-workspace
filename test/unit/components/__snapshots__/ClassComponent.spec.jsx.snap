// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassComponent should match enzyme shallow snapshot 1`] = `
<h3>
  Class Component
</h3>
`;
